std/algorithm/searching.d

```diff
@@ -2960,6 +2960,162 @@ unittest
     }
 }
 
+/**
+Iterates the passed range and selects the min and max element with `less`.
+If the extreme element occurs multiple time, the first occurrence will be
+returned.
+This function is more efficient than calling both $(LREF minElement) and
+$(LREF maxElement).
+
+Params:
+    map = custom accessor for the comparison key
+    selector = custom mapping for the extrema selection
+    seed = custom seed to use as initial element
+    r = Range from which the extreme value will be selected
+
+Returns:
+    The extreme value according to `map` and `selector` of the passed-in values.
+
+See_Also:
+    $(LREF minElement), $(LREF maxElement)
+*/
+auto minmaxElement(alias map = "a", alias selector = "a < b", Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range)
+in
+{
+    assert(!r.empty, "r is an empty range");
+}
+body
+{
+    alias mapFun = unaryFun!map;
+    alias Element = ElementType!Range;
+    Unqual!Element seed = r.front;
+    r.popFront();
+    return minmaxElement!(map, selector)(r, seed);
+}
+
+private auto minmaxElement(alias map = "a", alias selector = "a < b", Range,
```

Comment:
1. Why not to use common `map` instead?
2. Prototype with 2 seeds, one for max and one for min looks more useful
3. Why we need this function? is it faster then reduce!(min, max)?